### ISSUE TITLE
Possible bug with Tiwary reweighting + MPI

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#183]
I take note of this probable bug here so that we do not forget it.

In the last month a subtle error appeared on travis. Namely, regtest basic/rt67-mpi is sometime failing with the following error:

/home/travis/build/plumed/plumed2/regtest/basic/rt67-mpi/tmp
Run driver
Done. Here is the error file:
FAILURE
Diff for COLVAR:
1,7c1
< #! FIELDS time phi psi metad.bias metad.rbias metad.rct
< #! SET min_phi -pi
< #! SET max_phi pi
< #! SET min_psi -pi
< #! SET max_psi pi
<  0.000000  -1.2379   0.8942   0.0000   0.0000   0.0000
<  1.000000  -1.4839   1.0482   0.0000   0.0000   0.0089
---
>  1.000000  -1.4839   1.0482   0.0000   0.0000   0.0089

Notice that because of something strange in the "diff" command on travis, until couple of weeks ago the full diff was not displayed on the travis log (the message was "the two binary files are different", as if diff was not able to recognize that files where ASCII). Anyway, this is solved now and, whenever the produced files differs from the reference the diff is properly shown.

Now, the problem is that:
1. Failure is not reproducible. Something it happens, sometime it doesn't. I tried to rerun from travis the failing regtests (there is a "refresh-like" button) and typically they work then.
2. Only rt67-mpi is failing. This makes me suspect that problem is hidden somewhere in the combination of Pratyush reweight with MPI.
3. However, error shows up in writing a COLVAR file, which is done in a separate Action. And looks related to the File infrastructure, which is weird.

Perhaps it is a bug of the Travis machines. Anyway, the fact that the problem always comes out in the same test make me think there is something wrong in this specific part of the code.

**Comment by valsson** [comment]
I generated this regtest so I will take a look what could be the problem and let you know.

**Comment by GiovanniBussi** [comment]
OK thanks! Notice that in the v2.2 and master branch I just removed it (see log) to avoid failures on travis. Also notice that I was not able to reproduce the failure on my computer, and that also on travis it fails only from time to time.

**Comment by valsson** [comment]
I was able (by chance) to reproduce the failure on my laptop (Mac OS X with homebrew clang mpi compilers).

It seems that if the machine running the regtest is under heavy load the code will sometime fail to write the header and the first line at time 0.0000 properly and instead write some binary crap into the COLVAR file (@^@^@^..). The rest of the COLVAR file is OK and identical to the reference so there is no problem with Pratyush reweighting and MPI. Instead it seem to be some file output issue.

How often it fails seems to depend how many MPI processes are used for the test, if I used mpiprocs=4 I got something like 1 failure out of 20 runs while with mpiprocs=10 I got 7 failures out of 20 runs. Note that I only see failures with mpiprocs=10 if the computer is under heavy load. 

I don't know why this issue doesn't influence other MPI regtests, perhaps they are run with mpiprocs=2 where there is less likely to observe this behavior. Thus, maybe one way fix this issue is to set mpiprocs=2 for this regtest.
